src/lib/capacityEngine.js

```diff
@@ -26,40 +26,38 @@ function capacityEngine(store, productIds, time) {
     const firstUsableTrancheId = calculateTrancheId(gracePeriodExpiration);
     const firstUsableTrancheIndex = firstUsableTrancheId - firstActiveTrancheId;
 
-    const { capacityNXM, capacityUsedNXM } = productPools.reduce(
-      ({ capacityNXM, capacityUsedNXM }, pool) => {
+    const { capacityAvailableNXM, capacityUsedNXM } = productPools.reduce(
+      ({ capacityAvailableNXM, capacityUsedNXM }, pool) => {
         const { allocations, trancheCapacities } = pool;
 
-        const totalCapacity = trancheCapacities
-          .slice(firstUsableTrancheIndex)
-          .reduce((total, capacity) => total.add(capacity), Zero)
-          .mul(NXM_PER_ALLOCATION_UNIT);
-
-        const totalCapacityUsed = allocations.reduce((total, allocation) => total.add(allocation), Zero);
-
-        const initiallyUsedCapacity = allocations
-          .slice(0, firstUsableTrancheIndex)
-          .reduce((total, allocation) => total.sub(allocation), totalCapacityUsed)
-          .mul(NXM_PER_ALLOCATION_UNIT);
-
-        if (totalCapacity.gt(initiallyUsedCapacity)) {
-          capacityNXM = capacityNXM.add(totalCapacity).sub(initiallyUsedCapacity);
-        }
-
-        capacityUsedNXM = totalCapacityUsed.mul(NXM_PER_ALLOCATION_UNIT).add(capacityUsedNXM);
-
-        return { capacityNXM, capacityUsedNXM };
+        // sum up all allocations to get used capacity
+        const used = allocations.reduce((total, allocation) => total.add(allocation), Zero);
+
+        // traverse all tranches and sum up available capacity on a per-tranche basis
+        const available = trancheCapacities
+          .slice(firstUsableTrancheIndex) // skip unusable
+          .reduce((total, capacity, index) => {
+            // allocations may surpass total capacity in some scenarios
+            const free = bnMax(capacity.sub(allocations[index]), Zero);
```

Comment:
So if the firstUsableTranchIndex is on we will get the array without the first trancheCapacity so the indexes will change and we cannot use those indexes in the allocations because they will point to the element before that one.
example:
allocations = [1,2,3,4]
trancheCapacities = [1,2,3,4]
we slice the tranches array, now index 0 will point to 2 in tranches but to 1 in allocations